ambry-server/src/integration-test/java/com.github.ambry.server/ServerTestUtil.java

```diff
@@ -516,584 +516,550 @@ protected static void endToEndReplicationWithMultiNodeMultiPartitionTest(int int
+          || resp.getPartitionResponseInfoList().get(0).getErrorCode() == ServerErrorCode.Blob_Not_Found) {
+        Assert.assertTrue(blobsChecked.contains(blobIds.get(j)));
+      } else {
+        try {
+          BlobData blobData = MessageFormatRecord.deserializeBlob(resp.getInputStream());
+          byte[] blobout = new byte[(int) blobData.getSize()];
+          int readsize = 0;
+          while (readsize < blobData.getSize()) {
+            readsize += blobData.getStream().read(blobout, readsize, (int) blobData.getSize() - readsize);
+          }
+          Assert.assertArrayEquals(data, blobout);
+          if (testEncryption) {
+            assertNotNull("MessageMetadata should not have been null",
+                resp.getPartitionResponseInfoList().get(0).getMessageMetadataList().get(0));
+            assertArrayEquals("EncryptionKey mismatch", encryptionKey,
+                resp.getPartitionResponseInfoList().get(0).getMessageMetadataList().get(0).getEncryptionKey().array());
+          } else {
+            assertNull("MessageMetadata should have been null",
+                resp.getPartitionResponseInfoList().get(0).getMessageMetadataList().get(0));
+          }
+        } catch (MessageFormatException e) {
+          Assert.fail();
+        }
+      }
+
+      // get blob all
+      getRequest = new GetRequest(1, "clientid2", MessageFormatFlags.All, partitionRequestInfoList, GetOption.None);
+      channel1.send(getRequest);
+      stream = channel1.receive().getInputStream();
+      resp = GetResponse.readFrom(new DataInputStream(stream), clusterMap);
+      if (resp.getPartitionResponseInfoList().get(0).getErrorCode() == ServerErrorCode.Blob_Deleted
+          || resp.getPartitionResponseInfoList().get(0).getErrorCode() == ServerErrorCode.Blob_Not_Found) {
+        Assert.assertTrue(blobsChecked.contains(blobIds.get(j)));
+        blobsChecked.remove(blobIds.get(j));
+      } else {
+        try {
+          BlobAll blobAll = MessageFormatRecord.deserializeBlobAll(resp.getInputStream(), blobIdFactory);
+          byte[] blobout = new byte[(int) blobAll.getBlobData().getSize()];
+          int readsize = 0;
+          while (readsize < blobAll.getBlobData().getSize()) {
+            readsize += blobAll.getBlobData()
+                .getStream()
+                .read(blobout, readsize, (int) blobAll.getBlobData().getSize() - readsize);
+          }
+          Assert.assertArrayEquals(data, blobout);
+          if (testEncryption) {
+            Assert.assertNotNull("EncryptionKey should not ne null", blobAll.getBlobEncryptionKey());
+            Assert.assertArrayEquals("EncryptionKey mismatch", encryptionKey, blobAll.getBlobEncryptionKey().array());
+          } else {
+            Assert.assertNull("EncryptionKey should have been null", blobAll.getBlobEncryptionKey());
+          }
+        } catch (MessageFormatException e) {
```

Comment:
i don't know if there are more like this, but if this is the only one, can you let this exception through too (instead of the fail immediately after)?

Reply:
There are several, just in this file, and possibly in other server test and token test related files. I started on this but realized that this is best done in an isolated patch.